### PR TITLE
feat/VETS CPC-1248 fix delete vet by vet id

### PR DESCRIPTION
**Pull request by orell2j**
**JIRA:** link to jira ticket
https://champlainsaintlambert.atlassian.net/browse/CPC-1081?atlOrigin=eyJpIjoiMmU4Y2Q3MjNiMWI1NDM0YjlmZDdkNDc0MDBlMTUzOWUiLCJwIjoiaiJ9

## Context:
What is the ticket about and why are we doing this change.
## Does this PR change the .vscode folder in petclinic-frontend?:
If the PR changes the .vscode folder, explain why in detail because it should not. 
Be sure to include cgerard321 as a reviewer.

<span style="color:red">**Reviewers need to check for any changes to the 
.vscode folder and add a comment about it to their review comments.**</span>

## Changes
What are the various changes and what other modules do those changes affect.
This can be bullet point or sentence format.

-fixed the delete frontend code, it didn't delete the vet in the mongodb 
-fixed the bug where after you click delete vet, it goes to vet's details page

## Before and After UI (Required for UI-impacting PRs)

BEFORE:

image

Deleting "John Doe":

image

Reloading the page:

image

AFTER:

image

After deleting "John Doe": 

image

MongoDB after delete:

image

If this is a change to the UI, include before and after screenshots to show the differences.
If this is a new UI feature, include screenshots to show reviewers what it looks like. 
## Dev notes (Optional)
Specific technical changes that should be noted
## Linked pull requests (Optional)
Pull request links

### CODE REVIEW

**Review by bluement — APPROVED** [review]
thanks for commenting the code, looks good good pr

**Comment by hristoiv11** [comment]
You have commits from other services, which is not ok and it may confuse others. I suggest creating a new branch and pulling everything from the main. Put everything you have worked on in the new branch, commit it, and then create a new PR.

**Review by kyrusho — COMMENTED** [review]
After reviewing your pull request, I have noticed that it is missing context. You could edit your PR by giving us context about what is happening to give us an idea about what is being solved on the delete vet by vet id, what changes have been made, and why are these necessary. Is it because the method was malfunctioning? Is it because it was missing something?

Additionally, please be precise on whether you have made modifications on the .vscode folder or not, because you are not answering the question being posed on the default PR template which is the following, Does this PR change the .vscode folder in petclinic-frontend?

File: package-lock.json [hunk 1]
```diff
@@ -1,1503 +1,1753 @@
...
+        "picocolors": "^1.0.0"
+      },
       "engines": {
-        "node": ">=12"
+        "node": ">=6.9.0"
       }
     },
-    "node_modules/@esbuild/android-arm": {
-      "version": "0.20.2",
-      "resolved": "https://registry.npmjs.org/@esbuild/android-arm/-/android-arm-0.20.2.tgz",
```
Comment (bluement) [inline package-lock.json:71]: why did you change so many lines in this file?

**Review by bluement — COMMENTED** [review]
two files that you changed are exactly what needed to be changed, the package-lock.json has a lot of changes so i assume that you worked on the frontend in intellij instead of vsc? other than that the code is clean and well commented.

**Comment by orell2j** [comment]
> You have commits from other services, which is not ok and it may confuse others. I suggest creating a new branch and pulling everything from the main. Put everything you have worked on in the new branch, commit it, and then create a new PR.

Thank you for you suggestion, I'll note it for myself as a way to make a proper branch. It will not happen again and I'll make a new branch from the main.

**Review by Apoleh — COMMENTED** [review]
When you commit, please remember to write self-explanatory comments to be more clear about what is happening during your developing cycle.

**Review by Edwxd — COMMENTED** [review]
Please be sure to verify changes that are being made to config files before submiting a PR. The packages-lock.json was completly modified. Some dependencies verision are not the same.  The package name was "petclinic-frontend" to "champlain-frontend" on line 2 of the file. Overall too many big changes that could cause the project to break if approved. Recommend to break the branch and restart a new branch to be able to be up to date with main.

**Comment by orell2j** [comment]
> After reviewing your pull request, I have noticed that it is missing context. You could edit your PR by giving us context about what is happening to give us an idea about what is being solved on the delete vet by vet id, what changes have been made, and why are these necessary. Is it because the method was malfunctioning? Is it because it was missing something?
> 
> Additionally, please be precise on whether you have made modifications on the .vscode folder or not, because you are not answering the question being posed on the default PR template which is the following, Does this PR change the .vscode folder in petclinic-frontend?

Understood, I'll put more context in my new PR, I was missing the import deleteVet in the VetListTable.tsx and fixed the button Delete, so after clicking it doesn't go to the Vet Details page. 

In general, I'll note it and use those suggestions for my new PR.
